.github/workflows/test.yml

```diff
@@ -20,9 +20,9 @@ jobs:
       - uses: actions/cache@v3
         with:
           path: ~/.conan2
-          key: ${{runner.os}}-conan-${{ hashFiles('**/conanfile.py') }}
+          key: ubuntu-latest-conan-${{ hashFiles('**/conanfile.py') }}
           restore-keys: |
-            ${{ runner.os }}-conan-
+            ubuntu-latest-conan-
```

Comment:
This gets to use the same cache as publish.yml, because we're bumping up against how much storage Github gives us before they start deleting the old caches. Getting rid of the redundant copy saves nearly a gigabyte.

Reply:
One problem is that the Windows build is _not_ efficient despite the cache: build times are still 15 minutes (though caching brought it down from 2 hours). Mac still has a 4-minute build (down from 40 minutes). I strongly doubt we'll have any test that passes on one OS and fails on another; a build matrix seems like redundant work and extra delays for no benefit.

Reply:
This was news to me: https://chadgolden.com/blog/github-actions-hosted-windows-runners-slower-than-expected-ci-and-you.

It might be a bit challenging, but if we can get the working directory and conan directories (especially the cache we download) to be on the D: drive we might get real perf improvements.

You also mentioned that installing git and cmake took a great deal of time, which I think is worth investigating.

Reply:
Sure, Windows optimization would be nice. but I think that's a separate topic from this PR. File a Jira ticket, and we'll look at it later.

Reply:
Ah, that's a happy medium I could get behind! I'll comment out the Windows part with a TODO to uncomment it if the Windows build is tolerably fast.

Reply:
I've filed https://viam.atlassian.net/browse/RSDK-10636